### PR TITLE
feat: [draft] add logic in replication for handling rows/cells that may exceed thresholds

### PR DESCRIPTION
**Pull request by jhambleton**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java [hunk 1]
```diff
@@ -181,19 +223,38 @@ public Boolean call() {
...
+        // TODO - added for debugging REMOVE
+        /**
+         * Iterator<Cell> cellIt = cellsByRow.getValue().iterator(); while (cellIt.hasNext()) { Put
+         * put = new Put(cellsByRow.getKey().deepCopyToNewArray());
+         *
+         * <p>for (int i = 0; i < (100_000-1) && cellIt.hasNext(); i++) { put.add(cellIt.next()); }
+         *
+         * <p>table.put(put); }
+         */
+        List<RowMutations> rowMutations =
```
Comment (vermas2012) [inline hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java:236]: Doing single rowmutation in a batch is very inefficient. This is going to build replication lag as we are sending an RPC per put.

File: hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java [hunk 2]
```diff
@@ -181,19 +223,38 @@ public Boolean call() {
...
+            buildRowMutations(
```
Comment (vermas2012) [inline hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java:237]: When you create multiple row mutations, you are breaking the transaction.

File: hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/configuration/HBaseToCloudBigtableReplicationConfiguration.java [hunk 3]
```diff
@@ -54,6 +54,51 @@ private HBaseToCloudBigtableReplicationConfiguration() {}
...
+  public static final Integer DEFAULT_FILTER_LARGE_ROWS_THRESHOLD_IN_BYTES = 190 * 1024 * 1024;
+
+  /**
+   * Determines the size in bytes of the row mutations that should be logged and dropped when
+   * replicating to Bigtable. Default: Approximate row size accepted for batch mutation request.
+   */
+  public static final String MAX_CELLS_PER_MUTATION_KEY =
+      "google.bigtable.replication.max_cells_per_mutation";
+
+  public static final Integer DEFAULT_MAX_CELLS_PER_MUTATION = 100_000 - 1;
```
Comment (vermas2012) [inline hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/configuration/HBaseToCloudBigtableReplicationConfiguration.java:82]: check if this is per mutation or per row mutation?

File: hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapter.java [hunk 4]
```diff
@@ -142,6 +154,27 @@ public final List<Cell> adaptIncompatibleMutations(BigtableWALEntry walEntry) {
...
+                > conf.getInt(
+                    FILTER_LARGE_CELLS_THRESHOLD_IN_BYTES_KEY,
+                    DEFAULT_FILTER_LARGE_CELLS_THRESHOLD_IN_BYTES)) {
+          // Drop the cell as it exceeds the size to be filtered
+          incrementDroppedIncompatibleMutationsCellSizeExceeded();
+
+          LOG.warn(
+              "Dropping mutation, cell value length, "
+                  + cell.getValueLength()
+                  + ", exceeds filter length, "
```
Comment (vermas2012) [inline hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapter.java:169]: config name.

File: hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapter.java [hunk 5]
```diff
@@ -142,6 +154,27 @@ public final List<Cell> adaptIncompatibleMutations(BigtableWALEntry walEntry) {
...
+                  + conf.getInt(
+                      FILTER_LARGE_CELLS_THRESHOLD_IN_BYTES_KEY,
+                      DEFAULT_FILTER_LARGE_CELLS_THRESHOLD_IN_BYTES)
+                  + ", cell: "
+                  + cell);
```
Comment (vermas2012) [inline hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapter.java:174]: double check, should log row key, column family qualifier and value length.

**Comment by jhambleton** [comment]
thank you for the feedback. per discussion, will address these and open a new PR

**Pull request closed without merging**